addons/mail/static/src/core/common/thread_service.js

```diff
@@ -652,6 +655,20 @@ export class ThreadService {
 
     /**
      * @param {import("@mail/core/common/thread_model").Thread} thread
+     * @param {number} id
+     * @returns {Promise<Thread>}
```

Comment:
It doesn't look like this function returns a Promise, but just a thread?

Reply:
The function is `async` so it returns a Promise.

Reply:
Ho my bad you right :cry: